### ISSUE TITLE
Utilise Swagger.io for API Gateway documentation generation

### ISSUE THREAD

**Issue by marklawlor** [jaws-framework/JAWS#236]
There are already tools that can import a Swagger.json file into API Gateway (https://github.com/awslabs/aws-apigateway-importer), so we should be able to perform the reverse and convert the AWSM format into a Swagger.json file.

JAWS could have a `jaws document` command which builds a swagger.json file of all your module endpoints and uploads a Swagger UI site to the S3 bucket

**Comment by marklawlor** [comment]
Actually this should be a JAWS plugin, not part of the core library. I'll close this and create a plugin repo for this.

**Comment by JamieCressey** [comment]
+1 for this

**Comment by vmadman** [comment]
@yaochen Plugin list can be found here: https://www.npmjs.com/browse/keyword/serverless%20plugins

For clarity, the thread above discusses converting Serverless format(s) into Swagger specs.  This plugin seems to do that and, I think, it is the repo that was referred to above: https://www.npmjs.com/package/serverless-plugin-swagger-export

I made that clarification because I found this thread while looking for the inverse; converting Swaggerless specs into Serverless meta files.  I have not found any solutions along those lines so far.7

**Comment by vmadman** [comment]
I just found this, though, but it is incomplete and, apparently, has not been updated since the "JAWS" was renamed to "Severless": https://github.com/marklawlor/serverless-swagger
